src/components/test-utils/mocks/MockedCustomProvider.tsx

```
@@ -0,0 +1,23 @@
+import React, { ReactNode } from 'react';
+import type { MockedResponse } from '@apollo/client/testing';
+
+/**
+ * A mock implementation of Apollo's MockedProvider for testing purposes.
+ * Allows inspection of provided mocks through data attributes.
+ *
+ * @param props - Component properties
+ * @param props.children - Child elements to be wrapped by the provider
```

ESLint

[error] tsdoc-param-tag-with-invalid-name: The @param block should be followed by a valid parameter name: The identifier cannot non-word characters

(tsdoc/syntax)